### PR TITLE
Cart

### PR DESCRIPTION
**Pull request by Nishant-96**
this pr is for cart management feature.
- It also includes some styling changes.
- replaced links with navlink

### CODE REVIEW

File: src/pages/cart/components/cart-card.jsx [hunk 1]
```diff
@@ -1,35 +1,79 @@
...
           </div>
           <div className="cart-card-desc-counter">
-            <button className="btn btn-primary product-card-button">-</button>
-            <p>10</p>
-            <button className="btn btn-primary product-card-button">+</button>
+            <button
+              className="btn btn-primary product-card-button"
+              onClick={() => {
+                return dispatch({
+                  type: "CART_QTY_DECREMENTER",
```
Comment (znjs) [inline src/pages/cart/components/cart-card.jsx:36]: use constants to avoid typo bugs

File: src/pages/cart/components/cart-card.jsx [hunk 2]
```diff
@@ -1,35 +1,79 @@
...
+            }}
+          >
             Remove From Cart
           </button>
-          <button className="btn btn-outlined product-card-outlined-btn ">
+          <button
+            className="btn btn-outlined product-card-outlined-btn "
+            onClick={() => {
+              return dispatch({
+                type: "CART_TO_WISHLIST",
```
Comment (znjs) [inline src/pages/cart/components/cart-card.jsx:71]: use constants to avoid typo bugs

File: src/pages/product-listing/components/product-card.jsx [hunk 3]
```diff
@@ -19,7 +19,12 @@ function ProductCard({ product: productProps }) {
         </div>
         <div className="product-card-wrapper-details">
           <FavoriteIcon
-            className="product-card-like-button"
+            className={
+              productProps.inWishlist
+                ? "product-card-like-button product-card-liked"
+                : "product-card-like-button"
+            }
+            // className={productProps.inWishlist ? "product-card-liked" : null}
```
Comment (znjs) [inline src/pages/product-listing/components/product-card.jsx]: remove the unused code

File: src/pages/product-listing/components/product-card.jsx [hunk 4]
```diff
@@ -47,8 +52,16 @@ function ProductCard({ product: productProps }) {
             </span>
           </div>
         </div>
-        <button className="btn btn-primary product-card-button">
-          Add To Cart
+        <button
+          className="btn btn-primary product-card-button"
+          onClick={() => {
+            return dispatch({
+              type: "ADD_TO_CART",
```
Comment (znjs) [inline src/pages/product-listing/components/product-card.jsx:58]: use constants to avoid typo bugs

File: src/utils/reducerFunc.js [hunk 5]
```diff
@@ -74,6 +75,91 @@ function reducerFunc(state, action) {
         }),
       };
       break;
+
+    case "ADD_TO_CART":
+      state = {
+        ...state,
+        cartItems: state.cartItems.map((curr) => {
```
Comment (znjs) [inline src/utils/reducerFunc.js:82]: make a constants file and import all constants from it

**Review by ragini07 — COMMENTED** [review]
-looks good
-Preview working fine
-Good naming convention
-Suggestion : add badge for cart to show items on the go when you have the bandwidth

**Review by ragini07 — APPROVED** [review]

**Review by znjs — APPROVED** [review]
- Good naming conventions
- Easily understandable code

**Pull request merged**